R/mod_hierarchical_count_table.R

```diff
@@ -64,23 +65,26 @@ compute_events_table <- function(event_df, pop_df, hierarchy, group_var, subjid_
   checkmate::assert_subset(group_var, names(pop_df))
   checkmate::assert_string(subjid_var, min.chars = 1)
 
+  modified_pop_df <- pop_df |>
+    dplyr::mutate(dplyr::across(dplyr::all_of(group_var), ~ add_na_factor_level(.x, "<NA>")))
```

Comment:
What do you think about:

``` r
pop_df <- no_na_pop_df
no_na_pop_df[[group_var]] <- add_na_factor_level(no_na_pop_df[[group_var]])
```

My impression is that is easier to read with the same effect. Also `modified_pop_df` does not clarify what is the modification done.